### ISSUE TITLE
train epoch

### ISSUE THREAD

**Issue by long123524** [McGregorWwww/UCTransNet#16]
Very good work, but I have a question how much should epoch be appropriate?Does your proposed model apply to a smaller number of datasets?

**Comment by McGregorWwww** [comment]
Thanks for you attention.
1. The epoch is decided by the early-stopping strategy and we just set a large threshold, e.g., 2000 epochs, the actual number of epochs is about 200.
2. The MoNuSeg dataset used in our experiment has only 30 images for training which we think is quite a small dataset in semantic segmentation tasks. We do not test our model on smaller datasets, maybe you can try it.
Hope this helps.

**Comment by long123524** [comment]
Thanks for the response, I used 2,000 images with 256× 256 pixel size, trained 150 epochs, and used UNet as pre-training, but the result is not as good as Swin-UNet, how should I improve it? Moreover, my objects is to be larger, not that very small objects.

**Comment by McGregorWwww** [comment]
> 

Maybe you can try to add more layers and heads of the Transformer.

**Comment by long123524** [comment]
Thanks, but I still have a problem and I found that training the model from scratch results better than training the network using UNet as a pretrained model, why?  The UNet is set with the same parameters as the UCTransNet.  Moreover, I found increasing epoch improved the accuracy of results, How much appropriate is the epoch set up to?

**Comment by McGregorWwww** [comment]
> Thanks, but I still have a problem and I found that training the model from scratch results better than training the network using UNet as a pretrained model, why? The UNet is set with the same parameters as the UCTransNet. Moreover, I found increasing epoch improved the accuracy of results, How much appropriate is the epoch set up to?

1. The reasons are quite complicated. We think the different performances of ‘Pre-training’ strategy are depends on the difficulties of tasks. Our suggestion is that for some tasks like GlaS where the original U-Net can achieve a quite good performance, using the pre-trained weights may further improve the final performance, while for the relatively difficult tasks like MoNuSeg which has dense and tiny nucleus, the pre-trained weights may harm the final performance.
2. In our experiments, the number of epochs is decided by the early-stopping strategy so that the numbers of epochs are uncertain and are different among all experiments.
